Content.Shared/Chemistry/ReagentUnit.cs

```diff
@@ -0,0 +1,184 @@
+﻿using Robust.Shared.Interfaces.Serialization;
+using Robust.Shared.Serialization;
+using System;
+using System.Linq;
+
+namespace Content.Shared.Chemistry
+{
+    [Serializable]
+    public struct ReagentUnit : ISelfSerialize
+    {
+        private int _value;
+        private static readonly int Shift = 2;
+
+        public static ReagentUnit MaxValue => new ReagentUnit(int.MaxValue);
+
+        private decimal ShiftDown()
```

Comment:
Not sure whether you need the full precision of `decimal` when working with this. IIRC, `double` can accurately represent 32 bit-integers completely so that should be good enough. It'd help with performance quite a bit.

Reply:
done